### ISSUE TITLE
Create Tyk Dashboard library for common data access use cases

### ISSUE THREAD

**Issue by davegarvey** [TykTechnologies/backstage-tyk-entity-provider#14]
CRUD style functions for common data types - but really, it's only APIs needed for PoC.

To replace the stuff written directly in the various components. For example, it can replace the fetching code written in:
- [scaffolder action](https://github.com/TykTechnologies/portal-playground/blob/42990f682717512a757b0b72e0ce8f06e260aeca/packages/backend/src/plugins/scaffolder/actions/tyk-api-create.ts#L22)
- [entity provider](https://github.com/TykTechnologies/portal-playground/blob/42990f682717512a757b0b72e0ce8f06e260aeca/plugins/backend-catalog-tyk-entitiy-provider/backend-catalog-tyk-entity-provider.ts#L78)
- [Tyk plugin component](https://github.com/TykTechnologies/portal-playground/blob/42990f682717512a757b0b72e0ce8f06e260aeca/plugins/tyk/src/components/APIListComponent/APIListComponent.tsx#L56)

Basically anywhere we are doing `fetch await` against the Tyk Dashboard API.
